Bigtable/src/Connection/Grpc.php

```diff
@@ -211,39 +213,104 @@ public function deleteCluster(array $args)
      */
     public function createAppProfile(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $parent = $this->pluck('parent', $args);
+        $appProfile = $this->pluck('appProfile', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'createAppProfile'], [
+            $parent,
+            $this->pluck('appProfileId', $args),
+            $this->appProfileObject($appProfile),
+            $this->addResourcePrefixHeader($args, $parent)
+        ]);
+    }
+
+    /**
+     * @param array $args
+     * @return AppProfile
+     */
+    private function appProfileObject(array $args)
+    {
+        return $this->serializer->decodeMessage(
+            new AppProfile(),
+            $this->pluckArray([
+                'name',
+                'etag',
+                'description',
+                'singleClusterRouting',
+            ], $args)
+        );
     }
 
     /**
      * @param array $args
      */
     public function getAppProfile(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $name = $this->pluck('name', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'getAppProfile'], [
+            $name,
+            $this->addResourcePrefixHeader($args, $name)
+        ]);
     }
 
     /**
      * @param array $args
      */
     public function listAppProfiles(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $parent = $this->pluck('parent', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'listAppProfiles'], [
+            $parent,
+            $this->addResourcePrefixHeader($args, $parent)
+        ]);
     }
 
     /**
      * @param array $args
      */
     public function updateAppProfile(array $args)
     {
-        throw new \BadMethodCallException('This method is not implemented yet');
+        $name = $this->pluck('name', $args);
+        $appProfile = $this->pluck('appProfile', $args);
+        return $this->send([$this->bigtableInstanceAdminClient, 'updateAppProfile'], [
+            $this->appProfileObject($appProfile),
+            $this->updateMaskObject($appProfile),
+            $this->addResourcePrefixHeader($args, $name)
+        ]);
+    }
+
+    /**
+     * @param array $args
+     * @return FieldMask
+     */
+    private function updateMaskObject(array $args)
+    {
+        $mask = array_map([$this, 'serializerObject'], array_keys($args));
+        return $this->serializer->decodeMessage(
+            new FieldMask(),
+            ['paths' => $mask]
+        );
+    }
+
+    /**
+     * @param string $key
+     * @return Serializer
+     */
+    private function serializerObject($key)
```

Comment:
Rename this method to `convertToSnakeCase` or something descriptive of its action, and fix the return type (should be `string`).